File: VB/HowToAccessSchedulerControl.Module.Web/SchedulerController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.Scheduler.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.XtraScheduler
Imports DevExpress.Persistent.Base.General

Public Class SchedulerController
	Inherits DevExpress.ExpressApp.ViewController

	Public Sub New()
		MyBase.New()

		'This call is required by the Component Designer.
		InitializeComponent()
		RegisterActions(components) 

	End Sub

    Private Sub SchedulerController_Activated(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Activated
      If View.ObjectTypeInfo.Implements(Of IEvent)() Then
         AddHandler View.ControlsCreated, AddressOf View_ControlsCreated
      End If
    End Sub
    Private Sub View_ControlsCreated(ByVal sender As Object, ByVal e As EventArgs)
        ' Access the current List View
        Dim currentView As ListView = CType(View, ListView)
        ' Access the View's List Editor as a SchedulerListEditor
        Dim listEditor As SchedulerListEditor = CType(currentView.Editor, SchedulerListEditor)
        '  Access the LOist Editor's ScheudlerControl  
        Dim scheduler As ASPxScheduler = CType(listEditor.Scheduler, ASPxScheduler)
        ' Set the desired time to be visible
        If Not scheduler Is Nothing Then
            scheduler.Views.DayView.VisibleTime = _
               New TimeOfDayInterval(New TimeSpan(6, 0, 0), New TimeSpan(23, 0, 0))
        End If
    End Sub
End Class

File: CS/HowToAccessSchedulerControl.Module.Web/SchedulerController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.Scheduler.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.XtraScheduler;

using DevExpress.Persistent.Base.General;

namespace HowToAccessSchedulerControl.Module.Web {
   public partial class SchedulerController : ViewController {
      public SchedulerController() {
         InitializeComponent();
         RegisterActions(components);
      }

      private void SchedulerController_Activated(object sender, EventArgs e) {
         if (View.ObjectTypeInfo.Implements<IEvent>())
            View.ControlsCreated += new EventHandler(View_ControlsCreated);
      }
      void View_ControlsCreated(object sender, EventArgs e) {
         // Access the current List View
         ListView view = (ListView)View;
         // Access the View's List Editor as a SchedulerListEditor
         SchedulerListEditor listEditor = (SchedulerListEditor)view.Editor;
         //  Access the List Editor's SchedulerControl  
         ASPxScheduler scheduler = (ASPxScheduler)listEditor.Scheduler;
         // Set the desired time to be visible
         if (scheduler != null)
            scheduler.Views.DayView.VisibleTime =
               new TimeOfDayInterval(new TimeSpan(6, 0, 0), new TimeSpan(11, 0, 0));
      }
   }
}
